src/granite/querying.cr

```diff
@@ -2,153 +2,170 @@ module Granite::Querying
+    def exists?(**criteria : Granite::Columns::Type) : Bool
+      exists? criteria.to_h
+    end
 
-  # Returns `true` if a records exists that matches *criteria*, otherwise `false`.
-  def exists?(**criteria : Granite::Columns::Type) : Bool
-    exists? criteria.to_h
-  end
+    # :ditto:
+    def exists?(criteria : Granite::ModelArgs) : Bool
+      exec_exists *build_find_by_clause(criteria)
+    end
 
-  # :ditto:
-  def exists?(criteria : Granite::ModelArgs) : Bool
-    exec_exists *build_find_by_clause(criteria)
-  end
+    # count returns a count of all the records
+    def count : Int32
+      scalar "SELECT COUNT(*) FROM #{quoted_table_name}", &.to_s.to_i
+    end
 
-  # count returns a count of all the records
-  def count : Int32
-    scalar "SELECT COUNT(*) FROM #{quoted_table_name}", &.to_s.to_i
-  end
+    def exec(clause = "")
+      switch_to_writer_adapter
+      adapter.open(&.exec(clause))
+    end
 
-  def exec(clause = "")
-    switch_to_writer_adapter
-    adapter.open(&.exec(clause))
-  end
+    def query(clause = "", params = [] of Granite::Columns::Type, &)
+      switch_to_writer_adapter
+      adapter.open { |db| yield db.query(clause, args: params) }
+    end
 
-  def query(clause = "", params = [] of Granite::Columns::Type, &)
-    switch_to_writer_adapter
-    adapter.open { |db| yield db.query(clause, args: params) }
-  end
+    def scalar(clause = "", &)
+      switch_to_writer_adapter
+      adapter.open { |db| yield db.scalar(clause) }
+    end
 
-  def scalar(clause = "", &)
-    switch_to_writer_adapter
-    adapter.open { |db| yield db.scalar(clause) }
-  end
+    private def exec_exists(clause : String, params : Array(Granite::Columns::Type)) : Bool
+      self.adapter.exists? quoted_table_name, clause, params
+    end
 
-  private def exec_exists(clause : String, params : Array(Granite::Columns::Type)) : Bool
-    self.adapter.exists? quoted_table_name, clause, params
-  end
+    private def build_find_by_clause(criteria : Granite::ModelArgs)
+      keys = criteria.keys
+      criteria_hash = criteria.dup
 
-  private def build_find_by_clause(criteria : Granite::ModelArgs)
-    keys = criteria.keys
-    criteria_hash = criteria.dup
+      clauses = keys.map do |name|
+        if criteria_hash.has_key?(name) && !criteria_hash[name].nil?
+          matcher = "= ?"
+        else
+          matcher = "IS NULL"
+          criteria_hash.delete name
+        end
 
-    clauses = keys.map do |name|
-      if criteria_hash.has_key?(name) && !criteria_hash[name].nil?
-        matcher = "= ?"
-      else
-        matcher = "IS NULL"
-        criteria_hash.delete name
+        "#{quoted_table_name}.#{quote(name.to_s)} #{matcher}"
       end
 
-      "#{quoted_table_name}.#{quote(name.to_s)} #{matcher}"
+      {clauses.join(" AND "), criteria_hash.values}
     end
-
-    {clauses.join(" AND "), criteria_hash.values}
   end
+
+  {% if parse_type("Spec").resolve? %}
```

Comment:
Just remove this macro that wraps everything in favor of the macro inside of the method itself